modules/meshcloud-mca-service-principal/module.tf

```diff
@@ -0,0 +1,81 @@
+terraform {
+  required_version = "> 1.1"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.81.0"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "2.46.0"
+    }
+    azapi = {
+      source  = "Azure/azapi"
+      version = "1.13.1"
+    }
+  }
+}
+
+
+data "azurerm_billing_mca_account_scope" "mca" {
+  billing_account_name = var.billing_account_name
+  billing_profile_name = var.billing_profile_name
+  invoice_section_name = var.invoice_section_name
+}
+
+resource "azuread_application" "mca" {
+  display_name = var.service_principal_name
+}
+
+resource "azuread_service_principal" "mca" {
+  client_id = azuread_application.mca.client_id
+}
+
+data "azapi_resource_list" "billing_role_definitions" {
+  type                   = "Microsoft.Billing/billingAccounts/billingProfiles/invoiceSections/billingRoleDefinitions@2020-05-01"
+  parent_id              = data.azurerm_billing_mca_account_scope.mca.id
+  response_export_values = ["*"]
+}
+
+locals {
+  azure_subscription_creator_role_id = jsondecode(
+    data.azapi_resource_list.billing_role_definitions.output).value[
+    index(jsondecode(data.azapi_resource_list.billing_role_definitions.output).value[*].properties.roleName, "Azure subscription creator")
+  ].id
+}
+
+resource "azapi_resource_action" "add_role_assignment_subscription_creator" {
```

Comment:
nice workaround for https://github.com/hashicorp/terraform-provider-azurerm/issues/15211
can you put the hint to that into the code that this is the reason you do the "hacky" azapi stuff, so that in the future we may refactor it when a first party solution becomes available

Reply:
Added 👍